packages/api/src/hooks/p2p/useP2PPaymentMethods.ts

```diff
@@ -0,0 +1,38 @@
+import { useMemo } from 'react';
+import useAuthorize from '../useAuthorize';
+import { PAYMENT_METHOD_ICONS } from '../../constants';
+import useQuery from '../../useQuery';
+
+/** A custom hook that returns a list of P2P available payment methods **/
+const useP2PPaymentMethods = () => {
+    const { isSuccess } = useAuthorize();
+    const { data, ...rest } = useQuery('p2p_payment_methods', { options: { enabled: isSuccess } });
+    // Modify the data to add additional information.
+    const modified_data = useMemo(() => {
+        const p2p_payment_methods = data?.p2p_payment_methods;
+
+        if (!p2p_payment_methods) return undefined;
+
+        return Object.keys(p2p_payment_methods).map(key => {
+            const payment_method = p2p_payment_methods[key];
+            const fields = Object.keys(payment_method.fields).map(field_key => payment_method.fields[field_key]);
+
+            return {
+                ...payment_method,
+                /** Payment method field definitions. */
+                fields,
+                /** Icon for each payment method based on the type */
+                icon: PAYMENT_METHOD_ICONS[payment_method.type],
+                /** Payment method id */
+                id: key,
+            };
+        });
+    }, [data]);
+
+    return {
+        p2p_payment_methods: modified_data,
```

Comment:
As per our convention we always return data and we can rename it in our component if we want.

Reply:
Sure @thisyahlen-deriv, but could you please point me to a link where I can read these conventions?

Reply:
There were no documentation regarding this, the initial idea we had when developing these hooks is that the hooks/entities will not always be as per api/hook name. We may transform it to serve multiple purposes. So we let the user choose what they want from data in the component level. Just a quality of life thing that became a convention.

Reply:
Yeah @adrienne-deriv mentioned that to me